[Ready] Makes kitchen cold rooms actually cold (Metastation)

81Denton opened the pull request:
:cl: Denton
tweak: Metastation's kitchen cold room is now really cold! Make sure to wear warm clothing.
tweak: The kitchen clothing vendor now stocks two winter jackets by default.
tweak: Goats are now fine with temperatures as low as 180°K. This is so that Pete doesn't freeze to death inside the cold room.
code: Added a kitchen area subtype for cold rooms and adjusted chef CQC to work inside it (no gameplay changes).
/:cl:

You know how it's called a cold room, looks like a cold room but is actually just another storage room?

I made the Meta kitchen cold room spawn at -80°C and added a freezer to it. Originally I wanted to pick a more realistic temperature (like -25°C), but the way body temperature works means that you'd have to sit inside it for minutes to feel an effect.

Now, you can store bodies inside it, trap people and let them freeze to death, or lower the temperature even further and use it as a ghetto cryo chamber.

Also:
Kitchen clothing vendor gets two winter jackets and goats survive at lower temperatures (so Pete doesn't just die).


![yeap](https://user-images.githubusercontent.com/32391752/49668845-b167b100-fa5f-11e8-8e7d-1788ff31dfbd.PNG)

81Denton committed:
Makes kitchen cold rooms actually cold (Metastation)

81Denton commented on _maps/map_files/MetaStation/MetaStation.dmm:
```diff
@@ -49951,69 +49912,68 @@
 /obj/effect/turf_decal/stripes/line{
 	dir = 8
 	},
+/obj/machinery/atmospherics/components/binary/passive_gate{
```
This one keeps the cold room freezer from cooling the whole distro loop

kingoftheautism replied:
Or, you know, you could simply place pump like everywhere else. It's not like I'm against that passive gate tho

81Denton commented:
Forgot to mention: If this gets merged, I'll change the cold rooms of other maps as well.

coiax commented:
I don't know if decals have their own internal temperature, but if not, I'll have gibs not generate miasma in low temperatures in #41782.

Then provided the chef's room is refrigerated, it won't rot.

yoyobatty commented:
I wonder if we could put a fridge next to a pump that is outputting to the air vent in there or a space cooler.

81Denton commented:
@Tlaltecuhtli had the good idea of adding H/E pipes instead of a single vent, I'll play around with it when I have time to work on it

Steelpoint commented:
I swear someone did this before in the distant past. Good idea nonetheless.

MACIEKBAKI commented:
Is it at such a low temperature to deal damage to someone without proper (winter) clothing?

Therapist1 commented:
This wont cause immense spacewind when I open the door though, will it?

kingoftheautism commented:
@Therapist1 It shouldn't, since pressure is relatively same, only difference is temperature. Time to scavenge maints to get some space heaters

Grazyn commented:
Clothing vendor is inside the cold room so you're forced to suffer a bit of freeze damage before you can dispense a coat

81Denton commented:
@MACIEKBAKI Yeah, it will start hurting you after a few seconds. That's why I added winter jackets to the chef clothing vendor, so chefs can hang out in their workplace without freezing to death.

@Therapist1 I adjusted the cold room's mole count so it's also about 101.3 kPa, but at a much lower temperature. It shouldn't cause much space wind.

Anonmare commented:
Slightly worried this might cause some annoying temperature leak every time the door gets opened

PKPenguin321 commented:
How quickly do the affects of the temperature set in? It shouldn't be instant

81Denton commented:
I'm gonna replace the vent with H/E piping and do some testing. It might even be better to start it at warmer temperatures so it's not as dangerous as let's say telecomms.

Therapist1 commented:
Perhaps you could just install some tiny fans at the doors as to not let the air leak out into maint or the kitchen

coiax commented:
Tiny fans barely exist canonically, only to compensate for our lack of complex airlock interlocking air pumps

81Denton commented:
Turns out that H/E pipes work really well for regulating room temperatures. You can even replace the freezer with a heater and have yourself a nice BBQ.
I just need to remodel the layout a bit so it looks less ugly.

![capture](https://user-images.githubusercontent.com/32391752/49585333-5732f700-f95e-11e8-845e-d222825844f1.PNG)

81Denton committed:
Merge remote-tracking branch 'tgstation/master' into stay-frosty

81Denton committed:
H/E pipes aww yeah

81Denton commented:
Alright, the piping is now black so it clashes less with the H/E color scheme.
I also tested the damage this room does - about 15 damage if you hang around without warm clothing for a minute. This of course changes if you crank up the temperature.

ShizCalev commented on code/datums/martial/cqc.dm:
```diff
@@ -19,7 +19,7 @@
 /datum/martial_art/cqc/can_use(mob/living/carbon/human/H)
 	var/area/A = get_area(H)
-	if(just_a_cook && !(istype(A, /area/crew_quarters/kitchen)))
+	if(just_a_cook && (!(istype(A, /area/crew_quarters/kitchen)) || !(istype(A, /area/crew_quarters/kitchen/coldroom))))
```
if it's not a type of `/area/crew_quarters/kitchen`, or a type of `/area/crew_quarters/kitchen/coldroom`?

81Denton replied:
Ah, didn't know that istype includes subtypes as well.

81Denton committed:
Removes redundant istype check from cqc.dm

optimumtact commented:
You've answered nobody's concerns about temperature leakage into other rooms.

Go and fucking find out already.

81Denton commented:
@optimumtact From testing it looks like temperature leakage is no issue. I bolted the cold room airlock open for a minute in a row - temp of the surrounding kitchen tiles dropped to temps between ~2.5°C and 10°C. This was not low enough to cause any damage to nearby humans.

During rounds, the airlock won't be open for that long in a row since chefs don't want Pete to escape.

Bawhoppen commented:
Pete saves the day yet again.
